### DISCUSSION TITLE
Tabs Persist inside a modal?

### DISCUSSION THREAD

**Discussion by RPotkowaVegasystems** [filamentphp/filament#11496] [Help]
### Package

Panel builder

### Package Version

v3.2.31

### How can we help you?

I have a modal with my own form, inside there are some tabs. Is it possible to persist the state of the tabs? When i use "persistTabInQueryString", the query is correct set, but when i close/open the modal, the tab jumps to first tab.

**Comment by leandrocfe** [comment]
Maybe you can use [mountUsing](https://filamentphp.com/docs/3.x/actions/modals#executing-code-when-the-modal-opens) to get the query and set the current tab when the modal opens

**Comment by RPotkowaVegasystems** [comment]
OK, but how can i acccess on the request('tab'), i get null? And how to set the tab to the request('tab') value?

**Comment by RPotkowaVegasystems** [comment]
If i use the ->persistTab()->id('id') inside Tabs::make(), its works, and when you hit any tab, the local storage values are correct set. But when you close the modal, the value are set to 'undefined'

**Comment by RPotkowaVegasystems** [comment]
I did a small workaround! I have created my own tabs.blade.php. In the 

`Tabs::make()->view('filament.components.modal-tabs')->persistTabInQueryString('modal-tab')`

And in the tabs, i have changed some code:

`
if (! tab || ! tabs.includes(tab)) {
{{--tab = tabs[@js($getActiveTab()) - 1]--}}
const url = new URL(window.location.href)
tab = url.searchParams.get( @js($getTabQueryStringKey()) ) ?? tabs[@js($getActiveTab()) - 1]
}
`
